paddle/fluid/operators/mkldnn/range_mkldnn_op.cc

```diff
@@ -0,0 +1,122 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using framework::Tensor;
+
+template <typename T>
+class RangeMKLDNNHandler
+    : public platform::MKLDNNHandlerNoCachingT<T, dnnl::eltwise_forward,
+                                               dnnl::eltwise_backward> {
+ public:
+  RangeMKLDNNHandler(const T step, const int tensor_size,
+                     const dnnl::engine engine, platform::Place cpu_place)
+      : platform::MKLDNNHandlerNoCachingT<T, dnnl::eltwise_forward,
+                                          dnnl::eltwise_backward>(engine,
+                                                                  cpu_place) {
+    auto md = dnnl::memory::desc({tensor_size - RANGE_SHIFT},
+                                 platform::MKLDNNGetDataType<T>(),
+                                 dnnl::memory::format_tag::a);
+
+    this->AcquireForwardPrimitiveDescriptor(dnnl::prop_kind::forward_training,
+                                            dnnl::algorithm::eltwise_linear, md,
+                                            1.0f, RANGE_SHIFT * step);
+  }
+
+  std::shared_ptr<dnnl::memory> AcquireSrcMemory(framework::Tensor *out) {
+    T *ptr = out->data<T>();
+    return this->AcquireMemoryFromPrimitive(this->fwd_pd_->src_desc(), ptr);
+  }
+
+  std::shared_ptr<dnnl::memory> AcquireDstMemory(framework::Tensor *out) {
+    T *ptr = out->data<T>();
+    return this->AcquireMemoryFromPrimitive(this->fwd_pd_->dst_desc(),
+                                            ptr + RANGE_SHIFT);
+  }
+
+  static constexpr int RANGE_SHIFT = 64 / sizeof(T);  // AVX512 stores 64 bytes
```

Comment:
- RANGE_SHIFT is required by oneDNN?
- What about avx2 machines, maybe you need some condition for this kernel ?